src/Mvc/Mvc.Core/src/ModelBinding/Binders/DateOnlyModelBinderProvider.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System.Globalization;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding.Binders;
+
+/// <summary>
+/// An <see cref="IModelBinderProvider"/> for binding <see cref="DateOnly" /> and nullable <see cref="DateOnly"/> models.
+/// </summary>
+public class DateOnlyModelBinderProvider : IModelBinderProvider
+{
+    internal const DateTimeStyles SupportedStyles = DateTimeStyles.AllowWhiteSpaces;
+
+    /// <inheritdoc />
+    public IModelBinder? GetBinder(ModelBinderProviderContext context)
+    {
+        if (context == null)
+        {
+            throw new ArgumentNullException(nameof(context));
+        }
```

Comment:
```suggestion
        ArgumentNullException.ThrowIfNull(context, nameof(context));
```
no need to change this PR if you're just following existing pattern, but the API above is available in 6.0+ so all code of this pattern could be updated . It's considered as performant, since it inlines. might be worth an issue for someone to go through and change all these.

there's others eg ArgumentException.ThrowIfNullOrEmpty() and others coming.

Reply:
fwiw I looked and there's also ArgumentOutOfRangeException.ThrowIf
and various more proposed on ArgumentException in https://github.com/dotnet/runtime/issues/77749
and more on AOORE on https://github.com/dotnet/runtime/issues/69590

Reply:
oh, found https://github.com/dotnet/aspnetcore/issues/43482

Reply:
Thanks! I think it makes sense to make this change since last implementation of the existing pattern is a couple years old. I'll make the change just for the new files here and leave the changes in the other model binders for the resolution of #43482.